Code/Framework/AzToolsFramework/Tests/EditorViewportHelperTests.cpp

```diff
@@ -0,0 +1,247 @@
+
+        void SetUpEditorFixtureImpl() override
+        {
+            // Setup entity to use for EntityDebugDisplayEventBus and tests
+            AZ::Entity* entity1 = nullptr;
+            m_entityId = CreateDefaultEditorEntity("DebugHelpersEntity", &entity1);
+
+            AZ::Entity* entity2 = nullptr;
+            m_entityId2 = CreateDefaultEditorEntity("DebugHelpersEntity2", &entity2);
+
+            AzFramework::EntityDebugDisplayEventBus::MultiHandler::BusConnect(entity2->GetId());
+            AzFramework::EntityDebugDisplayEventBus::MultiHandler::BusConnect(entity1->GetId());
+
+            // DebugDisplay to use in DisplayHelpers
+            AzFramework::DebugDisplayRequestBus::BusPtr debugDisplayBus;
+            AzFramework::DebugDisplayRequestBus::Bind(debugDisplayBus, TestViewportId);
+            m_debugDisplay = AzFramework::DebugDisplayRequestBus::FindFirstHandler(debugDisplayBus);
+
+            m_cameraState = AzFramework::CreateDefaultCamera(AZ::Transform::CreateIdentity(), AzFramework::ScreenSize(1024, 768));
+
+            m_entityVisibleEntityDataCacheMock = AZStd::make_unique<::testing::NiceMock<MockEditorVisibleEntityDataCacheInterface>>();
+            m_editorHelpers = AZStd::make_unique<AzToolsFramework::EditorHelpers>(m_entityVisibleEntityDataCacheMock.get());
+            m_viewportSettings = AZStd::make_unique<ViewportSettingsTestImpl>();
+
+            m_viewportSettings->Connect(TestViewportId);
+            m_viewportSettings->m_helpersVisible = true;
+            m_viewportSettings->m_iconsVisible = true;
+
+            using ::testing::_;
+            using ::testing::Return;
+
+            EXPECT_CALL(*m_entityVisibleEntityDataCacheMock, GetVisibleEntityId(_))
+                .WillOnce(Return(m_entityId))
+                .WillOnce(Return(m_entityId2));
+            ON_CALL(*m_entityVisibleEntityDataCacheMock, VisibleEntityDataCount()).WillByDefault(Return(2));
+            ON_CALL(*m_entityVisibleEntityDataCacheMock, IsVisibleEntityVisible(_)).WillByDefault(Return(true));
+        }
+
+        void TearDownEditorFixtureImpl() override
+        {
+            m_viewportSettings->Disconnect();
+            m_viewportSettings.reset();
+            m_editorHelpers.reset();
+            m_entityVisibleEntityDataCacheMock.reset();
+            AzFramework::EntityDebugDisplayEventBus::MultiHandler::BusDisconnect();
+        }
+
+        // EntityDebugDisplayEventBus overrides ...
+        // This function is called in DisplayComponents which is responsible for drawing the helpers, if this is called it means that a
+        // helper has been drawn
+        void DisplayEntityViewport(const AzFramework::ViewportInfo&, AzFramework::DebugDisplayRequests&) override
+        {       
+            const AZ::EntityId entityId = *AzFramework::EntityDebugDisplayEventBus::GetCurrentBusId();
+
+            m_displayEntityViewportEvent = true;
+            m_drawCount += 1;
+            m_drawnEntity = entityId;
```

Comment:
update: As this can now be called multiple times, having an `AZStd::vector` that we push the entityIds back to might be better, and we can check if we got 1 or more in certain situations